pkg/instancetype/apply/cpu.go

```diff
@@ -0,0 +1,157 @@
+//nolint:gocyclo
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2024 Red Hat, Inc.
+ *
+ */
+package apply
+
+import (
+	k8sv1 "k8s.io/api/core/v1"
+	k8sfield "k8s.io/apimachinery/pkg/util/validation/field"
+
+	virtv1 "kubevirt.io/api/core/v1"
+	v1beta1 "kubevirt.io/api/instancetype/v1beta1"
+
+	preferenceApply "kubevirt.io/kubevirt/pkg/instancetype/preference/apply"
+)
+
+func applyCPU(
+	field *k8sfield.Path,
+	instancetypeSpec *v1beta1.VirtualMachineInstancetypeSpec,
+	preferenceSpec *v1beta1.VirtualMachinePreferenceSpec,
+	vmiSpec *virtv1.VirtualMachineInstanceSpec,
+) Conflicts {
+	if vmiSpec.Domain.CPU == nil {
+		vmiSpec.Domain.CPU = &virtv1.CPU{}
+	}
+
+	// If we have any conflicts return as there's no need to apply the topology below
+	if conflicts := validateCPU(field, instancetypeSpec, vmiSpec); len(conflicts) > 0 {
+		return conflicts
+	}
+
+	if vmiSpec.Domain.CPU.Model == "" && instancetypeSpec.CPU.Model != nil {
+		vmiSpec.Domain.CPU.Model = *instancetypeSpec.CPU.Model
+	}
```

Comment:
We allow users to override this, correct?

Reply:
Would appear so but I think that's a bug given this is in the instance type?! 

I'm going to track latent things like this in https://github.com/kubevirt/kubevirt/issues/13044 if it's okay and might address in follow up PRs.